typescript/cli/src/deploy/core.ts

```diff
@@ -138,10 +140,12 @@ export async function runCoreApply(params: ApplyParams) {
 
   switch (multiProvider.getProtocol(chain)) {
     case ProtocolType.Ethereum: {
+      // Assert EVM-specific addresses for strict type checking
+      const evmAddresses = deployedCoreAddresses as EvmCoreAddresses;
```

Comment:
can't we use the zod schemas for validating and parsing here to avoid the type assertion?